editoast/src/core/simulation.rs

```diff
@@ -96,51 +96,41 @@ impl PhysicsConsistParameters {
 }
 
 impl PhysicsConsistParameters {
-    pub fn compute_length(&self) -> u64 {
+    pub fn compute_length(&self) -> Length {
         let towed_rolling_stock_length = self
             .towed_rolling_stock
             .as_ref()
             .map(|trs| trs.length)
-            .unwrap_or(0.0);
-
-        let length = self
-            .total_length
-            .unwrap_or(self.traction_engine.length + towed_rolling_stock_length);
-
-        // Convert m to mm
-        let length = length * 1000.0;
+            .unwrap_or_default();
 
```

Comment:
When I see some `* 1000` disappear in this refactor, I'm thinking both "nice, we now **know**, thanks to the type system, that it's not the place to do it" 🚀 and "we're going to need a testing of the full OSRD stack once this PR is going to be merged to see if we communicate correctly with the `front` and `core`" 😨

Reply:
You are pointing the thing that worries me. In the database, with the macro, we should be quite confident that everything is coherent.

However, with the serde extension, I _suppose_ that by default it will consider mksa units. Maybe we should write explicit serialization/deserialization functions to be sure. And yes, testing by hand, a lot